### PR TITLE
Add CRI support for windows container/pod updates

### PR DESCRIPTION
**Pull request by katiewasnothere**
This PR adds support to CRI for requesting updates for windows containers/pods as well as LCOW containers/pods. 

This PR also pulls in a beta version of cri-api in order to get recent changes to the container update API to allow for passing windows container resources in requests.

### CODE REVIEW

**Comment by katiewasnothere** [comment]
@kevpar @ambarve @dcantah @anmaxvl fyi

File: pkg/server/container_update_resources.go [hunk 1]
```diff
@@ -30,90 +30,122 @@ import (
...
+			if status.State == sandbox.StateUnknown || status.State == sandbox.StateNotReady {
+				return status, errors.Errorf("sandbox %q must be running to update resources, instead has state %d", id, status.State)
+			}
+
+			return status, c.updateContainerResourcesV2(ctx, sndbx.Container, r)
+		}); err != nil {
+			return errors.Wrap(err, "failed to update resources")
+		}
+
+	} else if err != nil {
```
Comment (dcantah) [inline pkg/server/container_update_resources.go]: Nit: redundant check as if we got here we know the error isn't nil already and we're not checking for any specific error anymore

**Review by anmaxvl — APPROVED** [review]
lgtm

File: pkg/server/container_update_resources.go (hunk 1 above)
Comment (katiewasnothere) [reply pkg/server/container_update_resources.go]: Done

File: pkg/server/container_update_resources.go [hunk 2]
```diff
@@ -128,7 +159,27 @@ func updateContainerSpec(ctx context.Context, cntr containerd.Container, spec *r
 	return nil
 }
 
-// updateOCILinuxResource updates container resource limit.
+// updateContainerTask sends the Update request to the task with the requested resourcces and annotations to handle
```
Comment (dcantah) [inline pkg/server/container_update_resources.go]: resources*

File: pkg/server/container_update_resources.go [hunk 3]
```diff
@@ -30,90 +30,121 @@ import (
...
+
+			return status, c.updateContainerResourcesV2(ctx, sndbx.Container, r)
+		}); err != nil {
+			return errors.Wrap(err, "failed to update resources")
+		}
+
 	}
-	return &runtime.UpdateContainerResourcesResponse{}, nil
+
+	return errors.Wrap(err, "failed to find container")
```
Comment (dcantah) [inline pkg/server/container_update_resources.go]: This was hard to read for me, can we re-order these like the below?

```go
if err != nil {
    if err == store.ErrNotExist{
          ...
    }
    returns errors.Wrap(err, "failed to find container")
}
// err == nil logic here
```

Comment (katiewasnothere) [reply pkg/server/container_update_resources.go]: I kept this structure to mimic what was there previously. I can restructure if you'd like.

File: pkg/server/container_update_resources.go (hunk 2 above)
Comment (katiewasnothere) [reply pkg/server/container_update_resources.go]: God, you always get me :P

File: pkg/server/container_update_resources.go (hunk 3 above)
Comment (dcantah) [reply pkg/server/container_update_resources.go]: It makes more sense to me but that's fine with me to leave it also.

**Review by dcantah — APPROVED** [review]
LGTM

**Comment by dcantah** [comment]
LGTM... again 😆

**Pull request merged**